pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -976,23 +976,48 @@ self: super: {
   # https://github.com/haskell/hoopl/issues/50
   hoopl = dontCheck super.hoopl;
 
-  purescript =
-    let
-      purescriptWithOverrides = super.purescript.override {
-        # PureScript requires an older version of happy.
-        happy = self.happy_1_19_9;
-      };
-
-      # PureScript is built against LTS-13, so we need to jailbreak it to
-      # accept more recent versions of the libraries it requires.
-      jailBrokenPurescript = doJailbreak purescriptWithOverrides;
+  # The most recent version of purescript-cst (0.4.0.0) has version
+  # bounds for LTS-17, so we need to jailbreak it for LTS-18.
+  # doJailbreak can likely be removed when the next version of
+  # purescript-cst is released, since the version bounds have
+  # been updated for LTS-18.
+  purescript-cst = doJailbreak super.purescript-cst;
 
-      # Haddocks for PureScript can't be built.
-      # https://github.com/purescript/purescript/pull/3745
-      dontHaddockPurescript = dontHaddock jailBrokenPurescript;
-    in
-    # Generate shell completions
-    generateOptparseApplicativeCompletion "purs" dontHaddockPurescript;
```

Comment:
No more completion?

Reply:
Oh good catch, accidentally removed this!

Reply:
Rebased and added this back to https://github.com/NixOS/nixpkgs/pull/145310/commits/749f0c355e3f611a2740389275dd0141d6920b0f.